services/worker-manager/src/api.js

```diff
@@ -474,6 +474,67 @@ builder.declare({
   return res.reply(worker.serializable());
 });
 
+builder.declare({
+  method: 'post',
+  route: '/workers/:workerPoolId:/:workerGroup/:workerId',
+  name: 'updateWorker',
+  title: 'Update an existing Worker',
+  category: 'Workers',
+  stability: APIBuilder.stability.stable,
```

Comment:
Is this really `stable` on the first release?

Reply:
We really haven't paid much attention to these levels.  We have three:
 * "experimental" is typically used for "don't use this, we might kill it".  For example, the worker-info information in the queue has this stability level, and we do intend to remove it.  Ideally we wouldn't use this level very often!
 * "stable" is typically used for "normal" API methods.  Almost all methods have this stability, except `reregisterWorker`, and I'm not sure why that's the case.
 * "deprecated" is used for methods we'd rather users didn't call, but which are still used.

So I think "stable" is the correct choice here (and IMHO `reregisterWorker` should be "stable" too).

Reply:
It's possible that I set `reregisterWorker` to `experimental`. I always thought new API methods shouldn't be labeled as "stable" from the get go but I do understand your reasoning. I'm fine both ways.

Reply:
> We really haven't paid much attention to these levels. We have three:
> 
>     * "experimental" is typically used for "don't use this, we might kill it".  For example, the worker-info information in the queue has this stability level, and we do intend to remove it.  Ideally we wouldn't use this level very often!
> 
>     * "stable" is typically used for "normal" API methods.  Almost all methods have this stability, except `reregisterWorker`, and I'm not sure why that's the case.
> 
>     * "deprecated" is used for methods we'd rather users didn't call, but which are still used.
> 
> 
> So I think "stable" is the correct choice here (and IMHO `reregisterWorker` should be "stable" too).

The docs in https://github.com/taskcluster/taskcluster/tree/main/libraries/api#stability-levels suggest that `stable` is reserved for the situations when we are confident the API won't change. I'm not sure we benefit from making this a `stable` endpoint as we made need to tweak it after the first release. Perhaps we can mark it as `experimental` for the first release, and make it `stable` once we are sure it is fully suitable and requires no further modification?

Reply:
OK